pull/src/endpoint.h

```diff
@@ -30,7 +30,6 @@ class Endpoint {
  private:
   CivetServer& server_;
   const std::string uri_;
-  std::vector<std::weak_ptr<Collectable>> collectables_;
```

Comment:
Wasn't the whole point of supporting multiple endpoints that those could serve different collectables?

Reply:
As I understand, this PR preserves that functionality.

A single `Exposer` contains one or more `Endpoint` objects, depending on the `uri` parameter(s) used when calling `Exposer::RegisterCollectable`. And each of those `Endpoint` objects contains its own `MetricsHandler`, responsible for scraping the corresponding `Endpoint::collectables_` vector.

This change preserves the cardinality of those relationships, just moving the `collectables_` vector down a level, from `Endpoint` to `MetricsHandler`, where the locking can be done more in a more straightforward matter.

Reply:
@gjasny: Is there still any concern regarding the handling of multiple endpoints?